src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -599,48 +600,46 @@ private static bool HasApplicableConditionalMethod(OverloadResolutionResult<Meth
             var parameters = localFunction.Parameters;
             var methodResult = validResult.Result;
 
-            bool hasErrors = false;
-
-            for (int i = 0; i < args.Count; ++i)
+            // We're only in trouble if a dynamic argument is passed to the
+            // params parameter and is ambiguous at compile time between
+            // normal and expanded form i.e., there is exactly one dynamic
+            // argument to a params parameter
+            if (OverloadResolution.IsValidParams(localFunction) &&
+                args.Count == parameters.Length)
```

Comment:
I don't think the second condition should be met.Consider this for example:
```
    static void Test4(int x = 0, params int[] y)
    {
        dynamic z = x;
        Test4(y: z);
    }
```
Should probably add a unit-test for similar scenario.